java/src/main/java/ai/rapids/cudf/Scalar.java

```diff
@@ -589,6 +618,11 @@ public int hashCode() {
       case STRING:
         valueHash = Arrays.hashCode(getUTF8());
         break;
+      case LIST:
+        try (ColumnView v = getListAsColumnView()) {
+          valueHash = v.hashCode();
```

Comment:
This should compute the hash of the values, not just the metadata.  Like Java's `String` we may want to cache the hashcode to avoid recomputing it over and over.

Reply:
Yes, I agree. The issue #8090 was filed for this.